ops/jujuversion.py

```diff
@@ -0,0 +1,75 @@
+import re
+from functools import total_ordering
+
+
+@total_ordering
+class JujuVersion:
+
+    PATTERN = r'^(\d{1,9})\.(\d{1,9})((?:\.|-([a-z]+))(\d{1,9}))?(\.(\d{1,9}))?$'
+
+    def __init__(self, version):
+        m = re.match(self.PATTERN, version)
+        if not m:
+            raise RuntimeError(f"{version} is not a valid Juju version string.")
+
+        self.major = int(m.group(1))
+        self.minor = int(m.group(2))
+        tag = m.group(4)
+        if tag is None:
+            tag = ''
+        self.tag = tag
+
+        patch = m.group(5)
+        if patch is None:
+            self.patch = 0
+        else:
+            self.patch = int(patch)
+
+        build = m.group(7)
+        if build is None:
+            self.build = 0
+        else:
+            self.build = int(build)
+
+    def __repr__(self):
+        if self.tag:
+            s = f'{self.major}.{self.minor}-{self.tag}{self.patch}'
+        else:
+            s = f'{self.major}.{self.minor}.{self.patch}'
+        if isinstance(self.build, int) and self.build > 0:
```

Comment:
Given the above code, isn't build always an int?
This could also just be simplified (because of python booleans) to

if self.build:

Reply:
Right, looks like I forgot to remove this check from my previous iteration on this.

Thanks for noticing.